File: sources/LRQ 1.0/net/ccbluex/liquidbounce/ui/client/GuiWelcome.java
/*
 * Decompiled with CFR 0.152.
 * 
 * Could not load the following classes:
 *  kotlin.jvm.internal.Intrinsics
 *  org.lwjgl.input.Keyboard
 *  org.lwjgl.opengl.GL11
 */
package net.ccbluex.liquidbounce.ui.client;

import java.awt.Color;
import kotlin.jvm.internal.Intrinsics;
import net.ccbluex.liquidbounce.LiquidBounce;
import net.ccbluex.liquidbounce.api.minecraft.client.gui.IFontRenderer;
import net.ccbluex.liquidbounce.api.minecraft.client.gui.IGuiButton;
import net.ccbluex.liquidbounce.api.util.WrappedGuiScreen;
import net.ccbluex.liquidbounce.features.module.Module;
import net.ccbluex.liquidbounce.features.module.modules.render.ClickGUI;
import net.ccbluex.liquidbounce.ui.client.GuiMainMenu;
import net.ccbluex.liquidbounce.ui.font.Fonts;
import net.ccbluex.liquidbounce.utils.MinecraftInstance;
import org.lwjgl.input.Keyboard;
import org.lwjgl.opengl.GL11;

public final class GuiWelcome
extends WrappedGuiScreen {
    @Override
    public void initGui() {
        this.getRepresentedScreen().getButtonList().add(MinecraftInstance.classProvider.createGuiButton(1, this.getRepresentedScreen().getWidth() / 2 - 100, this.getRepresentedScreen().getHeight() - 40, "Ok"));
    }

    @Override
    public void drawScreen(int mouseX, int mouseY, float partialTicks) {
        this.getRepresentedScreen().drawBackground(0);
        IFontRenderer font = Fonts.font35;
        font.drawCenteredString("Thank you for downloading and installing our client!", (float)this.getRepresentedScreen().getWidth() / 2.0f, (float)this.getRepresentedScreen().getHeight() / 8.0f + (float)70, 0xFFFFFF, true);
        font.drawCenteredString("Here is some information you might find useful if you are using LiquidBounce for the first time.", (float)this.getRepresentedScreen().getWidth() / 2.0f, (float)this.getRepresentedScreen().getHeight() / 8.0f + (float)70 + (float)font.getFontHeight(), 0xFFFFFF, true);
        font.drawCenteredString("\u00a7lClickGUI:", (float)this.getRepresentedScreen().getWidth() / 2.0f, (float)this.getRepresentedScreen().getHeight() / 8.0f + (float)80 + (float)(font.getFontHeight() * 3), 0xFFFFFF, true);
        StringBuilder stringBuilder = new StringBuilder().append("Press ");
        Module module = LiquidBounce.INSTANCE.getModuleManager().get(ClickGUI.class);
        if (module == null) {
            Intrinsics.throwNpe();
        }
        font.drawCenteredString(stringBuilder.append(Keyboard.getKeyName((int)module.getKeyBind())).append(" to open up the ClickGUI").toString(), (float)this.getRepresentedScreen().getWidth() / 2.0f, (float)(this.getRepresentedScreen().getHeight() / 8) + 80.0f + (float)(font.getFontHeight() * 4), 0xFFFFFF, true);
        font.drawCenteredString("Right-click modules with a '+' next to them to edit their settings.", (float)this.getRepresentedScreen().getWidth() / 2.0f, (float)this.getRepresentedScreen().getHeight() / 8.0f + (float)80 + (float)(font.getFontHeight() * 5), 0xFFFFFF, true);
        font.drawCenteredString("Hover a module to see it's description.", (float)this.getRepresentedScreen().getWidth() / 2.0f, (float)this.getRepresentedScreen().getHeight() / 8.0f + (float)80 + (float)(font.getFontHeight() * 6), 0xFFFFFF, true);
        font.drawCenteredString("\u00a7lImportant Commands:", (float)this.getRepresentedScreen().getWidth() / 2.0f, (float)this.getRepresentedScreen().getHeight() / 8.0f + (float)80 + (float)(font.getFontHeight() * 8), 0xFFFFFF, true);
        font.drawCenteredString(".bind <module> <key> / .bind <module> none", (float)this.getRepresentedScreen().getWidth() / 2.0f, (float)this.getRepresentedScreen().getHeight() / 8.0f + (float)80 + (float)(font.getFontHeight() * 9), 0xFFFFFF, true);
        font.drawCenteredString(".autosettings load <name> / .autosettings list", (float)this.getRepresentedScreen().getWidth() / 2.0f, (float)this.getRepresentedScreen().getHeight() / 8.0f + (float)80 + (float)(font.getFontHeight() * 10), 0xFFFFFF, true);
        font.drawCenteredString("\u00a7lNeed help? Feel free to contact us!", (float)this.getRepresentedScreen().getWidth() / 2.0f, (float)this.getRepresentedScreen().getHeight() / 8.0f + (float)80 + (float)(font.getFontHeight() * 12), 0xFFFFFF, true);
        font.drawCenteredString("YouTube: https://youtube.com/ccbluex", (float)this.getRepresentedScreen().getWidth() / 2.0f, (float)this.getRepresentedScreen().getHeight() / 8.0f + (float)80 + (float)(font.getFontHeight() * 13), 0xFFFFFF, true);
        font.drawCenteredString("Twitter: https://twitter.com/ccbluex", (float)this.getRepresentedScreen().getWidth() / 2.0f, (float)this.getRepresentedScreen().getHeight() / 8.0f + (float)80 + (float)(font.getFontHeight() * 14), 0xFFFFFF, true);
        font.drawCenteredString("Forum: https://forum.ccbluex.net/", (float)this.getRepresentedScreen().getWidth() / 2.0f, (float)this.getRepresentedScreen().getHeight() / 8.0f + (float)80 + (float)(font.getFontHeight() * 15), 0xFFFFFF, true);
        super.drawScreen(mouseX, mouseY, partialTicks);
        GL11.glScalef((float)2.0f, (float)2.0f, (float)2.0f);
        Fonts.font40.drawCenteredString("Welcome!", (float)(this.getRepresentedScreen().getWidth() / 2) / 2.0f, (float)this.getRepresentedScreen().getHeight() / 8.0f / (float)2 + (float)20, new Color(0, 140, 255).getRGB(), true);
    }

    @Override
    public void keyTyped(char typedChar, int keyCode) {
        if (1 == keyCode) {
            return;
        }
        super.keyTyped(typedChar, keyCode);
    }

    @Override
    public void actionPerformed(IGuiButton button) {
        if (button.getId() == 1) {
            MinecraftInstance.mc.displayGuiScreen(MinecraftInstance.classProvider.wrapGuiScreen(new GuiMainMenu()));
        }
    }
}

File: sources/LiquidSense Dev/net/ccbluex/liquidbounce/ui/client/GuiWelcome.kt
/*
 * LiquidBounce Hacked Client
 * A free open source mixin-based injection hacked client for Minecraft using Minecraft Forge.
 * https://github.com/CCBlueX/LiquidBounce/
 */
package net.ccbluex.liquidbounce.ui.client

import net.ccbluex.liquidbounce.LiquidBounce
import net.ccbluex.liquidbounce.features.module.modules.render.ClickGUI
import net.ccbluex.liquidbounce.ui.font.Fonts
import net.minecraft.client.gui.GuiButton
import net.minecraft.client.gui.GuiScreen
import org.lwjgl.input.Keyboard
import org.lwjgl.opengl.GL11
import java.awt.Color

class GuiWelcome : GuiScreen() {

    override fun initGui() {
        this.buttonList.add(GuiButton(1, this.width / 2 - 100, height - 40, "Ok"))
    }

    override fun drawScreen(mouseX: Int, mouseY: Int, partialTicks: Float) {
        drawBackground(0)

        val font = Fonts.font35

        font.drawCenteredString( LiquidBounce.CLIENT_NAME + "B" +  LiquidBounce.CLIENT_VERSION, width / 2F - 1, height / 8F + 70, 0xffffff, true)

        font.drawCenteredString("By CCBlueX Develop By AquaVit", width / 2F - 6, height / 8F + 80F + font.FONT_HEIGHT * 2, 0xffffff, true)
        font.drawCenteredString("本客户端是免费的请勿倒卖以及购买", width / 2F - 13, height / 8F + 80F + font.FONT_HEIGHT * 3, 0xffffff, true)
        font.drawCenteredString("群785835501持续更新 仅供学习勿用与商业", width / 2F - 23, height / 8 + 80F + font.FONT_HEIGHT * 4, 0xffffff, true)
        font.drawCenteredString("最终解释权归AquaVit所有 已经遵循GPLV3协议", width / 2F - 26, height / 8F + 80F + font.FONT_HEIGHT * 5, 0xffffff, true)
        font.drawCenteredString("如果使用过程中发现BUG 请反馈QQ2924270322", width / 2F - 26, height / 8F + 80F + font.FONT_HEIGHT * 6, 0xffffff, true)
        font.drawCenteredString("2020/8/18 UPDATE", width / 2F - 3, height / 8F + 80F + font.FONT_HEIGHT * 7, 0xffffff, true)

        super.drawScreen(mouseX, mouseY, partialTicks)

        // Title
        GL11.glScalef(2F, 2F, 2F)
        Fonts.font40.drawCenteredString("爷懒得写说明", width / 2 / 2F, height / 8F / 2  - 2, Color(0, 140, 255).rgb, true)
    }

    override fun keyTyped(typedChar: Char, keyCode: Int) {
        if (Keyboard.KEY_ESCAPE == keyCode)
            return

        super.keyTyped(typedChar, keyCode)
    }

    override fun actionPerformed(button: GuiButton) {
        if (button.id == 1) {
            mc.displayGuiScreen(GuiMainMenu())
        }
    }
}